File: app/src/main/java/com/android/testdai/application/ui/settings/SettingsFragment.java
package com.android.testdai.application.ui.settings;

import android.content.Intent;
import android.os.Bundle;
import android.preference.Preference;
import android.preference.PreferenceFragment;
import android.preference.SwitchPreference;
import android.support.annotation.Nullable;

import com.android.testdai.R;
import com.android.testdai.application.ui.dialog.DialogImage;
import com.android.testdai.application.ui.dialog.DialogImageCache;
import com.android.testdai.util.PreferencesUtil;
import com.squareup.picasso.Picasso;
import com.squareup.picasso.PicassoProvider;

import static com.android.testdai.util.Constants.APP_PREFERENCES_DOUBLE_CLICK;
import static com.android.testdai.util.Constants.APP_PREFERENCES_ERROR_LIMIT;
import static com.android.testdai.util.Constants.APP_PREFERENCES_TIME_LIMIT;

public class SettingsFragment extends PreferenceFragment {

    private final String DIALOG_IMAGE_CACHE = "DIALOG_IMAGE_CACHE";

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        addPreferencesFromResource(R.xml.preferences);

        SwitchPreference timeLimit = (SwitchPreference) findPreference("time_limit");
        timeLimit.setChecked(PreferencesUtil.getInstance(getActivity()).getPreference(APP_PREFERENCES_TIME_LIMIT));
        timeLimit.setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
            @Override
            public boolean onPreferenceClick(Preference preference) {
                PreferencesUtil.getInstance(getActivity()).setPreference(APP_PREFERENCES_TIME_LIMIT, timeLimit.isChecked());
                return true;
            }
        });

        SwitchPreference errorLimit = (SwitchPreference) findPreference("error_limit");
        errorLimit.setChecked(PreferencesUtil.getInstance(getActivity()).getPreference(APP_PREFERENCES_ERROR_LIMIT));
        errorLimit.setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
            @Override
            public boolean onPreferenceClick(Preference preference) {
                PreferencesUtil.getInstance(getActivity()).setPreference(APP_PREFERENCES_ERROR_LIMIT, errorLimit.isChecked());
                return true;
            }
        });

        SwitchPreference doubleClick = (SwitchPreference) findPreference("double_click");
        doubleClick.setChecked(PreferencesUtil.getInstance(getActivity()).getPreference(APP_PREFERENCES_DOUBLE_CLICK));
        doubleClick.setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
            @Override
            public boolean onPreferenceClick(Preference preference) {
                PreferencesUtil.getInstance(getActivity()).setPreference(APP_PREFERENCES_DOUBLE_CLICK, doubleClick.isChecked());
                return true;
            }
        });

        Preference shareIt = (Preference) findPreference("share_it");
        shareIt.setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
            @Override
            public boolean onPreferenceClick(Preference preference) {

                Intent i = new Intent (Intent.ACTION_SEND);
                i.setType("text/plain");
                i.putExtra(Intent.EXTRA_TEXT, getString(R.string.share_text)+getString(R.string.app_source));
                i = Intent.createChooser(i, getString(R.string.send_share));
                startActivity(i);

                return true;
            }
        });

  /*      Preference cacheImage = (Preference) findPreference("image_cache");
        cacheImage.setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
            @Override
            public boolean onPreferenceClick(Preference preference) {

                DialogImageCache dialog = DialogImageCache.newInstance(true);
                dialog.show(getFragmentManager(), DIALOG_IMAGE_CACHE);

                return true;
            }
        });

        Preference clearCache = (Preference) findPreference("clear_cache");
        clearCache.setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
            @Override
            public boolean onPreferenceClick(Preference preference) {

                DialogImageCache dialog = DialogImageCache.newInstance(false);
                dialog.show(getFragmentManager(), DIALOG_IMAGE_CACHE);

                return true;
            }
        });
*/
    }

}

File: app/src/main/java/com/android/testdai/ui/fragments/SettingsFragment.kt
package com.android.testdai.ui.fragments

import android.content.Intent
import android.os.Bundle
import android.preference.SwitchPreference
import android.view.View
import androidx.preference.Preference
import androidx.preference.PreferenceFragmentCompat
import androidx.preference.SwitchPreferenceCompat
import com.android.testdai.R
import com.android.testdai.di.components.DaggerScreenComponent
import com.android.testdai.managers.SharedPreferencesManager
import com.android.testdai.ui.activities.BaseActivity
import com.android.testdai.utils.Constants.*
import com.android.testdai.utils.PreferencesUtil
import javax.inject.Inject

class SettingsFragment: PreferenceFragmentCompat() {

    @Inject
    lateinit var sharedPreferencesManager: SharedPreferencesManager

    private val TIME_LIMIT = "time_limit"
    private val ERROR_LIMIT = "error_limit"
    private val DOUBLE_CLICK = "double_click"
    private val SHARE_ID = "share_it"


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        inject()
    }

    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        setPreferencesFromResource(R.xml.preferences, rootKey);
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val timeLimit = findPreference<SwitchPreferenceCompat>(TIME_LIMIT)
        timeLimit?.isChecked = sharedPreferencesManager.isTimeLimit
        timeLimit?.setOnPreferenceClickListener { preference ->
            sharedPreferencesManager.isTimeLimit = timeLimit.isChecked
            true
        }

        val errorLimit = findPreference<SwitchPreferenceCompat>(ERROR_LIMIT)
        errorLimit?.isChecked = PreferencesUtil.getInstance(activity).getPreference(APP_PREFERENCES_ERROR_LIMIT)
        errorLimit?.setOnPreferenceClickListener { preference ->
            sharedPreferencesManager.isErrorLimit = errorLimit.isChecked
            true
        }

        val doubleClick = findPreference<SwitchPreferenceCompat>(DOUBLE_CLICK)
        doubleClick?.isChecked = PreferencesUtil.getInstance(activity).getPreference(APP_PREFERENCES_DOUBLE_CLICK)
        doubleClick?.setOnPreferenceClickListener { preference ->
            sharedPreferencesManager.isDoubleClick = doubleClick.isChecked
            true
        }

        val shareIt = findPreference<Preference>(SHARE_ID)
        shareIt?.setOnPreferenceClickListener { preference ->

            var i = Intent(Intent.ACTION_SEND)
            i.type = "text/plain"
            i.putExtra(Intent.EXTRA_TEXT, getString(R.string.share_text) + getString(R.string.app_source))
            i = Intent.createChooser(i, getString(R.string.send_share))
            startActivity(i)

            true
        }
    }

    private fun inject() {
        DaggerScreenComponent.builder()
                .applicationComponent((activity as BaseActivity).applicationComponent)
                .build()
                .inject(this)
    }

}